From b966bb98e07de6273d00cfcc04c5ca2986e73b86 Mon Sep 17 00:00:00 2001
From: Don Zickus <dzickus@redhat.com>
Date: Fri, 12 Sep 2014 18:06:32 -0400
Subject: [powerpc] boot: Don't change link address for OF-based platforms

Message-id: <1410545655-205645-164-git-send-email-dzickus@redhat.com>
Patchwork-id: 94249
O-Subject: [RHEL7 PATCH 163/626] powerpc/boot: Don't change link address for OF-based platforms
Bugzilla: 1127366
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Steve Best <sbest@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: David Gibson <dgibson@redhat.com>

https://bugzilla.redhat.com/show_bug.cgi?id=1127366

commit f5467e28d4e2569b2c71435d438ce3c7f9ac238e
Author: Paul Mackerras <paulus@samba.org>
Date:   Tue Oct 29 16:21:26 2013 +1100

    powerpc/boot: Don't change link address for OF-based platforms

    Commit c55aef0e5bc6 ("powerpc/boot: Change the load address for the
    wrapper to fit the kernel") adjusts the wrapper address unnecessarily
    for platforms that use arch/powerpc/boot/of.c, since the code there
    allocates space for the kernel wherever it can find it and doesn't
    necessarily load the kernel at address 0.  Changing the link address
    is actually harmful since it can cause the zImage to overlap with
    Open Firmware and thus fail to boot.

    To fix this, we set make_space to n for all of the platforms that
    use of.o.

    Signed-off-by: Paul Mackerras <paulus@samba.org>
    Signed-off-by: Benjamin Herrenschmidt <benh@kernel.crashing.org>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/powerpc/boot/wrapper b/arch/powerpc/boot/wrapper
index 25a61ef..c7e74a7 100755
--- a/arch/powerpc/boot/wrapper
+++ b/arch/powerpc/boot/wrapper
@@ -153,18 +153,22 @@ of)
 pseries)
     platformo="$object/of.o $object/epapr.o"
     link_address='0x4000000'
+    make_space=n
     ;;
 maple)
     platformo="$object/of.o $object/epapr.o"
     link_address='0x400000'
+    make_space=n
     ;;
 pmac|chrp)
     platformo="$object/of.o $object/epapr.o"
+    make_space=n
     ;;
 coff)
     platformo="$object/crt0.o $object/of.o $object/epapr.o"
     lds=$object/zImage.coff.lds
     link_address='0x500000'
+    make_space=n
     pie=
     ;;
 miboot|uboot*)
-- 
1.7.1